### PR TITLE
Listings API written.  Sorry, this one's a doozy

### PR DESCRIPTION
**Pull request by ahusu**

### CODE REVIEW

File: server/controllers/index.js [hunk 1]
```diff
@@ -1,24 +1,62 @@
-const model = require('../models');
+const model = require("../models");
 
 module.exports = {
-
-  getListings: (req, res) => {
-    res.end();
+  getProducts: (req, res) => {
```
Comment (pnilan) [inline server/controllers/index.js:4]: I think we'll need both a "getListings" function and a "getProducts" function.

Comment (pnilan) [reply server/controllers/index.js:4]: getListings for populating the "marketplace". and getProducts for populating the listing page/create listing form.

File: server/models/index.js [hunk 2]
```diff
@@ -1,24 +1,158 @@
...
+      L.condition,
+      L.price,
+      L.description,
+      L.status,
+      L.created_at`;
+    return await db.query(query, [id]);
   },
 
-  createListingInDB: () => {
+  getMyListingsFromDB: async (id) => {
```
Comment (pnilan) [inline server/models/index.js:72]: Oh I see you have a "getMyListings" function here. I think we'll still need a general getListings though.

File: server/routes.js [hunk 3]
```diff
@@ -2,19 +2,25 @@ var controller = require('./controllers');
 var router = require('express').Router();
 
 // GET /api/listings?query=
-router.get('/listings', controller.getListings);
+router.get('/api/products', controller.getProducts);
```
Comment (pnilan) [inline server/routes.js]: The /api/ portion of the route should have already been handled in server.js

**Review by pnilan — CHANGES_REQUESTED** [review]
See comments.

File: server/routes.js (hunk 3 above)
Comment (pnilan) [reply server/routes.js]: Can you update the function in server.js to be /api/pl (to represent products & listings API) and then all the URLs in the routes.js file can be /listings, /products, etc

File: server/controllers/index.js (hunk 1 above)
Comment (ahusu) [reply server/controllers/index.js:4]: I've talked with Holly and Aaron, and I think we're good with using get Products to populate marketplace

File: server/routes.js (hunk 3 above)
Comment (ahusu) [reply server/routes.js]: Yep got it.

**Review by muzammilwaheedisme — COMMENTED** [review]
lgtm

**Pull request merged**